sdks/java/core/src/main/java/org/apache/beam/sdk/io/CountingInput.java

```diff
@@ -118,6 +118,12 @@ public static UnboundedCountingInput unbounded() {
   /**
    * A {@link PTransform} that will produce a specified number of {@link Long Longs} starting from
    * 0.
+   *
+   * <pre>{@code
+   * Pipeline p = ...
+   * PTransform<PBegin, PCollection<Long>> producer = CountingInput.upTo(10L);
+   * PCollection<Long> bounded = p.apply(producer);
```

Comment:
Suggest simplifying as:

```java
PCollection<Long> bounded = p.apply(CountingInput.upTo(10L));
```

Usually, we don't save a `PTransform` in a local variable unless we want to reuse it or build it up over many lines of code – we normally just apply it inline.